pycbc/workflow/core.py

```diff
@@ -135,6 +133,11 @@ def __init__(self, cp, name, ifos=None, out_dir=None, tags=None,
             The folder to store output files of this job.
         tags : list of strings
             A list of strings that is used to identify this job.
+        reuse_executable: boolean [default True]
+            Set pegasus_name to name (rather than tagged_name)?
```

Comment:
This help text isn't really useful. Let me try and explain and hopefully this can be converted into something useful!

You will often have workflows that creates multiple `Executable` objects for the same actual executable (for e.g. if doing injection runs you'll likely have a new Executable object for pycbc_multi_inspiral for every injection run). From pegasus' perspective these are then different actual executables. This might result in unnecessary copies (although this is not relevant for any practical PyCBC setup) and will result in pegasus' logging treating them as separate files (the logging can be useful for tracking run times). It can also result in pegasus not batching jobs if you are asking for some executables to be "clustered".

So if this option is *on* the executable name sent to pegasus is the name of the executable itself and nothing else. This then means that pegasus thinks that this is one executable and will deal with it accordingly.

If it is *off* the executable name sent to pegasus is the tagged_name, which will mean pegasus thinks they are different executables. A useful use case for this is pycbc_inspiral jobs in different ifos and/or different injection runs. These might have different run times and so having these separated when examining workflow performance can be useful.

Reply:
Thanks, @spxiwh: see if what I wrote as a summary is good enough.

Reply:
Looks good. I think this needs rebasing because of test failures (although there may now be more test failures).